File: LeetCode/Easy/0228-summary-ranges/0228-summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        int i = 0;
        int len = nums.size();
        vector<string> ret;
        while(i < len){
            int low = i;
            int high = i;

            while(high + 1 < len && nums[high]+1 == nums[high+1]){
                ++high;
            }
            
            if (low == high) ret.push_back(to_string(nums[low]));
            else ret.push_back(to_string(nums[low]) + "->" + to_string(nums[high]));

            i = high + 1;
        }
        return ret;
    }
};

File: LeetCode/Easy/0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        i = 0
        length = len(nums)
        ret = []
        while i < length:
            low = i
            
            while i + 1 < length and nums[i+1] == nums[i]+1:
                i += 1
            
            if low == i:
                ret.append(str(nums[low]))
            else:
                ret.append(str(nums[low]) + "->" + str(nums[i]))
                
            i += 1
        return ret
        